cedar-policy/src/api.rs

```diff
@@ -2341,6 +2354,73 @@ impl FromStr for PolicySet {
 }
 
 impl PolicySet {
+    /// Desesrialize the policy set from a JSON value
+    pub fn from_json_value(v: serde_json::Value) -> Result<Self, PolicySetError> {
+        let est: est::PolicySet = serde_json::from_value(v)?;
+        Self::from_est(est)
+    }
+
+    /// Build the policy set AST from the EST
+    fn from_est(est: est::PolicySet) -> Result<Self, PolicySetError> {
+        let mut pset = Self::default();
+
+        for PolicyEntry { id, policy } in est.templates {
+            let template = Template::from_est(Some(PolicyId(id)), policy)?;
+            pset.add_template(template)?;
+        }
+
+        for PolicyEntry { id, policy } in est.static_policies {
+            let p = Policy::from_est(Some(PolicyId(id)), policy)?;
+            pset.add(p)?;
+        }
+
+        for Link { id, target, env } in est.links {
+            let env = env
+                .into_iter()
+                .map(|(key, value)| (key.into(), EntityUid(value)))
+                .collect();
+            pset.link(PolicyId(target), PolicyId(id), env)?;
+        }
+
+        Ok(pset)
+    }
+
+    /// Deserialize the [`PolicySet`] from a JSON string
+    pub fn from_json_str(src: impl AsRef<str>) -> Result<Self, PolicySetError> {
+        let est: est::PolicySet = serde_json::from_str(src.as_ref())?;
+        Self::from_est(est)
+    }
```

Comment:
I think in other places we have the full trifecta of json constructors -- from value, str, and Read